src/main/java/org/ballerinalang/plugins/idea/runconfig/BallerinaRunConfigurationProducerBase.java

```diff
@@ -144,6 +195,26 @@ private void setConfigurations(@NotNull BallerinaApplicationConfiguration config
         configuration.setWorkingDirectory(workingDirectory);
     }
 
+    private void setTestConfigurations(@NotNull BallerinaTestConfiguration configuration, @NotNull PsiFile file,
+                                       @NotNull String packageInFile) {
+        configuration.setPackage(packageInFile);
+        // Set the working directory. If this is not set, package in sub modules will not run properly.
+        Module module = ModuleUtilCore.findModuleForPsiElement(file);
+        if (module == null) {
+            return;
+        }
+        VirtualFile moduleFile = module.getModuleFile();
+        if (moduleFile == null) {
+            return;
+        }
+        String workingDirectory = moduleFile.getParent().getPath();
+        if (workingDirectory.endsWith(".idea")) {
+            workingDirectory = workingDirectory.replace(".idea", "");
```

Comment:
Shall we use BallerinaConstants.IDEA_CONFIG_DIRECTORY or any other constant  for this?